home-page-script.js

```diff
@@ -154,28 +113,69 @@ function goBackToHomePage() {
 const searchInput = document.getElementById("search-input");
 const searchButton = document.getElementById("search-button");
 const searchResult = document.getElementById("search-result");
+const ingredientPriceInfo = document.getElementById("ingredient-price-info");
 
 //
 function displaySearchedRecipes() {
   const keyword = searchInput.value.trim().toLowerCase();
   searchResult.innerHTML = `Search result for ${keyword}:`;
+  const searchTitleCheckbox = document.getElementById("search-title");
+  const searchIngredientCheckbox = document.getElementById("search-ingredient");
   searchInput.value = "";
   recipeContainer.innerHTML = "";
+  ingredientPriceInfo.innerHTML = "";
 
   if (!keyword) {
     goBackToHomePage();
     return;
   }
 
-  const filteredRecipes = recipesObjectArray.filter((recipe) => {
-    return recipe.title.toLowerCase().includes(keyword);
-  });
-  if (!filteredRecipes.length) {
-    recipeContainer.innerHTML = "No recipes found :(";
-    return;
+  let filteredRecipe = [];
+
+  if (searchTitleCheckbox.checked) {
+    filteredRecipe = filteredRecipe.concat(
+      recipesObjectArray.filter((recipe) =>
+        recipe.title.toLowerCase().includes(keyword)
+      )
+    );
+  }
+
+  //if ingredient search checkbox is checked
+  if (searchIngredientCheckbox.checked) {
+    const matchedIngredientRecipes = recipesObjectArray.filter((recipe) =>
+      recipe.ingredients.some((ingredient) =>
+        ingredient.name.toLowerCase().includes(keyword)
+      )
+    );
+
+    filteredRecipe = filteredRecipe.concat(matchedIngredientRecipes);
+
+    const matchedIngredientObject = ingredientPriceArray.find(
+      (element) => element.name.toLowerCase()===(keyword)
+    );
+
+  
+    if (matchedIngredientObject) {
+      ingredientPriceInfo.innerHTML = `${matchedIngredientObject.name} : ${matchedIngredientObject.price}`;
+    } 
+
+    const uniqueID = new Set();
```

Comment:
It works but I think you can get more benefits from the `Set` by initialising it with an array of IDs.

```js
const uniqueIDs = new Set(filteredRecipe.map((recipe) => {
  return recipe.id
});
const uniqueRecipes = filteredRecipe.filter((recipe) => {
  return uniqueIDs.has(recipe.id);
});
```